doc/application-maintainer.rst

```diff
@@ -0,0 +1,141 @@
+==========================================
+Maintaining Neptune-Versioned Applications
+==========================================
+
+This guide is intended to help having a common ground when versioning
+applications, in particular applications that are deployed to provide services,
+but it could probably be extended to other type of applications. The goal is to
+let users and people affected by changes in applications quickly know what to
+expect on new releases.
+
+Since applications have very different impact and some are more critical than
+others, it is very hard to have a strict guideline like with `libraries
+<library-maintainer.rst>`_, so this is just a guideline and each project must
+adapt it to what works best for it.
+
+.. contents::
+
+
+Version Numbers
+===============
+
+Numbering versions of releases should follow `SemVer 2.0 <http://semver.org/>`_.
+This only means, a version number should have the form::
+
+  X.Y.Z[-P][+M]
+
+Where:
+
+X
+  is the **Major** version
+Y
+  is the **Minor** version
+Z
+  is the **Patch** version
+P
+  is a pre-release identifier (optional)
+M
+  is build metadata (optional)
+
+For more details on which kind of characters can be used on each, please
+consult the SemVer guide. The meaning of these numbers, though, does not
+strictly match the SemVer guide.
+
+
+General Guideline
+=================
+
+These are the most general guidelines presented, and ideally all applications
+should follow these principles.
+
+Since for applications the most important thing to know on new releases is how
+risky it is to deploy the new release, version numbers should indicate that:
+how likely it is that something breaks after a release.
+
+Patch Releases
+--------------
+Patch releases should include only minimal changes, and be close to zero risk to
+deploy.
+
+People interacting with or affected by the application should be able to ignore
+any patch releases. No other applications should need to be updated to be able
+to deploy a new patch release.
+
+Minor Releases
+--------------
+Minor releases should be comparatively low risk. People interacting with or
+affected by the application should be notified and should care about the
+update, but it should be safe to assume that it is very unlikely that something
+could break because of the upgrade.
+
+Major Releases
+--------------
+Major releases should be made only when major breaking changes, refactoring or
+rewrites occur. Major versions come with a higher risk of breakage, and since
+they normally carry breaking changes affecting other applications, they
+usually can't be rolled back without also rolling back other applications.
+Because of this, they should be closely monitored and not only by the
+maintainers but also by people interacting with or affected by the application.
+
+Special major version 0 should only be used for new applications being deployed
```

Comment:
> Wording suggestion: "The special major version number 0".

:+1: 

>"0.x.x" ?

Technically the major version is 0 :)
Maybe "The special major version number 0 (the v0.x.x series) ..."?